src/Arcus.Templates.Tests.Integration/Fixture/InMemorySecretProvider.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Arcus.Security.Secrets.Core.Interfaces;
+using GuardNet;
+
+namespace Arcus.Templates.Tests.Integration.Fixture
+{
+    /// <summary>
+    /// <see cref="ISecretProvider"/> implementation that provides an in-memory storage of secrets by name.
+    /// </summary>
+    public class InMemorySecretProvider : ICachedSecretProvider
+    {
+        private readonly IDictionary<string, string> _secretValueByName;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="InMemorySecretProvider"/> class.
+        /// </summary>
+        /// <param name="secretValueByName">The sequence of combinations of secret names and values.</param>
+        public InMemorySecretProvider(params (string name, string value)[] secretValueByName)
```

Comment:
You know I'm not a fan of this 😁

Can't we just us a dictionary given you use `ToDictionary` later on anyway?

Reply:
Yep, will do.